File: frontend/src/objects/review/ReviewAppealIdentity.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type {
  CustomerId,
  OrderId,
  PersonName,
  ReviewAppealId,
  RiderId,
  StoreId,
} from '@/objects/core/SharedObjects'

export type ReviewAppealIdentity = {
  id: ReviewAppealId
  orderId: OrderId
  customerId: CustomerId
  customerName: PersonName
  storeId: StoreId
  riderId?: RiderId
}

File: backend/src/main/scala/services/review/objects/ReviewAppealIdentity.scala
package services.review.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given
import services.rider.objects.*
import services.order.objects.*
import services.merchant.objects.*
import services.customer.objects.*

import system.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class ReviewAppealIdentity(
    id: ReviewAppealId,
    orderId: OrderId,
    customerId: CustomerId,
    customerName: PersonName,
    storeId: StoreId,
    riderId: Option[RiderId],
)
object ReviewAppealIdentity:
  given Encoder[ReviewAppealIdentity] = deriveEncoder
  given Decoder[ReviewAppealIdentity] = deriveDecoder
